orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/pipeline/strategy/DetermineSourceServerGroupTask.groovy

```diff
@@ -49,6 +51,15 @@ class DetermineSourceServerGroupTask implements RetryableTask {
 
   @Override
   TaskResult execute(StageExecution stage) {
+    ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
+    if (stage.getContext() != null && stage.getContext().get("configFiles") != null
+        && (!((List) stage.getContext().get("configFiles")).isEmpty())) {
+      Object yamlObj = yamlReader.readValue(stage.getContext().get("configFiles").get(0), Object.class);
+      if (yamlObj.get("metadata") != null && yamlObj.get("metadata").get("labels") != null
+          && yamlObj.get("metadata").get("labels").get("cloud.googleapis.com/location") != null) {
+        stage.getContext().put("region", yamlObj.get("metadata").get("labels").get("cloud.googleapis.com/location"))
+      }
+    }
```

Comment:
I'm not convinced this is the right place for this. `DetermineSourceServerGroups` is a generic task used by lots of different providers, so having custom stuff for one provider in it seems weird.

Can we not extend this and override the `execute` method for Cloudrun?